code/game/machinery/shieldgen.dm

```diff
@@ -553,3 +553,656 @@
+			dy += 1
+			t1 += dy
+			t2 = t1 - dx
+			if(t2 > 0)
+				t1 = t2
+				dx -= 1
+
+		if(exterior_only)
+			for(var/turf/target_tile as anything in list_of_turfs)
+				if (!(target_tile in inside_shield) && isspaceturf(target_tile) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
+					var/obj/structure/emergency_shield/modular/deploying_shield = new(target_tile)
+					deploying_shield.shield_generator = src
+					deployed_shields += deploying_shield
+			calculate_regeneration()
+			active_power_usage += deployed_shields.len * BASE_MACHINE_ACTIVE_CONSUMPTION * 0.1
+			return
+
+		for(var/turf/target_tile as anything in list_of_turfs)
+			if (!(target_tile in inside_shield) && isopenturf(target_tile) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
+				var/obj/structure/emergency_shield/modular/deploying_shield = new(target_tile)
+				deploying_shield.shield_generator = src
+				deployed_shields += deploying_shield
+		calculate_regeneration()
+		active_power_usage += deployed_shields.len * BASE_MACHINE_ACTIVE_CONSUMPTION * 0.1
+		return
+
+	//this code only runs on radius less than 10 and gives us a more accurate circle that is more compatible with decimal values
+	var/list/inside_shield = circle_range_turfs(src, radius - 1)//in the future we might want to apply an effect to the turfs inside the shield
+	if(exterior_only)
+		for(var/turf/target_tile as anything in circle_range_turfs(src, radius))
+			if (!(target_tile in inside_shield) && isspaceturf(target_tile) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
+				var/obj/structure/emergency_shield/modular/deploying_shield = new(target_tile)
+				deploying_shield.shield_generator = src
+				deployed_shields += deploying_shield
+		calculate_regeneration()
+		active_power_usage += deployed_shields.len * BASE_MACHINE_ACTIVE_CONSUMPTION * 0.1
+		return
+
+	for(var/turf/target_tile as anything in circle_range_turfs(src, radius))
+		if (!(target_tile in inside_shield) && isopenturf(target_tile) && !(locate(/obj/structure/emergency_shield/modular) in target_tile))
+			var/obj/structure/emergency_shield/modular/deploying_shield = new(target_tile)
+			deploying_shield.shield_generator = src
+			deployed_shields += deploying_shield
+	calculate_regeneration()
+	active_power_usage += deployed_shields.len * BASE_MACHINE_ACTIVE_CONSUMPTION * 0.1
+
+
+
+/obj/machinery/modular_shield_generator/Destroy()
+	QDEL_LIST(deployed_shields)
+	active_power_usage = BASE_MACHINE_ACTIVE_CONSUMPTION * 0.5
+	return ..()
+
+/obj/machinery/modular_shield_generator/update_icon_state()
+
+	icon_state = ("gen_[!(machine_stat & NOPOWER)?"[recovering ?"recovering_":"ready_"]":"no_power_"][(panel_open)?"open":"closed"]")
+	return ..()
+
+//ui stuff
+/obj/machinery/modular_shield_generator/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Modularshieldgen")
```

Comment:
yeah this is why that UI thing was failing, it's because you didn't update it here?

Reply:
when i changed that to camel case it said it couldnt find a ui named modularShieldGen

when i only changed the export to camel case it said Modularshieldgen had no export.

Reply:
found a proper solution, leaving it unresolved incase you dont like it though.